### DISCUSSION TITLE
Smart Chat errors - "No Smart Connections found", "API key is invalid", 429

### DISCUSSION THREAD

**Discussion by belsandre** [brianpetro/obsidian-smart-connections#694] [Q&A] [answered]
Hi, when I first installed Smart Connections months ago, Smart Chat worked. But in the recent months, it has stopped working. I'd love to debug it:

When I click "Save" on the OpenAI API Key field, I would get popups that say "No Smart Connections found" and "API key is invalid", but the API key is correct... When I go to OpenAI platform, it shows the key as having been used today.

When I try to chat something, I would get the error "API Error. See console logs for details."

Below is a screenshot of my console log.
<img width="253" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/6519097/960fc943-4375-40c5-b4d6-0e250d7f8e16">

**Comment by brianpetro** [comment, marked as answer]
Hi @belsandre 

That error means you have to add credit to your OpenAI account 🌴

**Reply by belsandre** [reply]
ah thank you...

**Comment by Ramadas108** [comment]
What about free llama 3.1 model, that should not use any credits? I get also API key is invalid on this model.

**Comment by ttodosi** [comment]
having trouble with new 3.1 ollama, can someone post a screenshot of config?

**Reply by brianpetro** [reply]
![Screenshot 2024-08-17 at 3 17 58 PM](https://github.com/user-attachments/assets/aae77389-3cee-48cc-8765-4e920efb025c)

**Comment by newbiespawn** [comment]
![Screenshot 2024-08-18 032049](https://github.com/user-attachments/assets/dbc76781-b8ed-4018-8bd2-843bf6e89f91)
Im having issues with not being able to select a Gemini Pro Model anymore on the latest update...

**Reply by brianpetro** [reply]
Hi @newbiespawn, it looks like an API key issue.

Once you add a valid API key, the "Refresh Model List" button will disappear, and the available models will be displayed in a dropdown menu 🌴

**Reply by brianpetro** [reply]
PS- cool note that you're working on 💡🌴

**Comment by newbiespawn** [comment]
Hey Brian, First of all THANK YOU for the plugin! I would love to help out.
I could not change anything (the drop down broke) if I tried setting embeds
first, I needed to change to Gemini pro WITHOUT vision first.

Good news it's working after uninstall and setting chat model first then
embeds.

**Comment by newbiespawn** [comment]
And thanks 🙂‍↕️

**Comment by MatthewDanh** [comment]
I am looking everywhere for an unsubscribe button, and I cannot find it. I sent an email to the legal department. However I'm checking here just in case I missed it somewhere.

**Reply by brianpetro** [reply]
The link to manage Smart Connect subscriptions is in the welcome email that contains your connection key 🌴
